modules/mongodb/charts.go

```diff
@@ -0,0 +1,490 @@
+	Units: "MiB",
+	Fam:   "memory",
+	Ctx:   "mongodb.memory",
+	Dims: module.Dims{
+		{ID: "mem.resident", Name: "resident"},
+		{ID: "mem.virtual", Name: "virtual"},
+		{ID: "mem.mapped", Name: "mapped"},
+		{ID: "mem.mappedWithJournal", Name: "mapped with journal"},
+	},
+}
+
+var chartPageFaults = module.Chart{
+	ID:    "page_faults",
+	Title: "Page faults",
+	Units: "page faults/s",
+	Fam:   "page_faults",
+	Ctx:   "mongodb.page_faults",
+	Dims: module.Dims{
+		{ID: "extra_info.page_faults", Name: "Page Faults", Algo: module.Incremental},
+	},
+}
+
+var chartAsserts = module.Chart{
+	ID:    "asserts",
+	Title: "Asserts",
+	Units: "asserts/s",
+	Fam:   "asserts",
+	Ctx:   "mongodb.asserts",
+	Type:  module.Stacked,
+	Dims: module.Dims{
+		{ID: "asserts.regular", Name: "regular", Algo: module.Incremental},
+		{ID: "asserts.warning", Name: "warning", Algo: module.Incremental},
+		{ID: "asserts.msg", Name: "msg", Algo: module.Incremental},
+		{ID: "asserts.user", Name: "user", Algo: module.Incremental},
+		{ID: "asserts.tripwire", Name: "tripwire", Algo: module.Incremental},
+		{ID: "asserts.rollovers", Name: "rollovers", Algo: module.Incremental},
+	},
+}
+
+// option charts
+var chartTransactionsCurrent = module.Chart{
+	ID:    "transactionsCurrent",
+	Title: "Current Transactions",
+	Units: "transactions",
+	Fam:   "transactionsCurrent",
+	Ctx:   "mongodb.transactionsCurrent",
+	Dims: module.Dims{
+		{ID: "transactions.currentActive", Name: "current active"},
+		{ID: "transactions.currentInactive", Name: "current inactive"},
+		{ID: "transactions.currentOpen", Name: "current open"},
+		{ID: "transactions.currentPrepared", Name: "current prepared"},
+	},
+}
+
+var chartGlobalLockActiveClients = module.Chart{
+	ID:    "globalLockActiveClients",
+	Title: "Active Clients",
+	Units: "clients",
+	Fam:   "clients",
+	Ctx:   "mongodb.currentQueue",
+	Dims: module.Dims{
+		{ID: "globalLock.activeClients.readers", Name: "readers"},
+		{ID: "globalLock.activeClients.writers", Name: "writers"},
+	},
+}
+
+var chartCollections = module.Chart{
+	ID:    "catalogStats",
+	Title: "Catalog Stats",
+	Units: "objects",
+	Fam:   "catalogStats",
+	Ctx:   "mongodb.catalogStats",
+	Type:  module.Stacked,
+	Dims: module.Dims{
+		{ID: "catalogStats.collections", Name: "collections"},
+		{ID: "catalogStats.capped", Name: "capped"},
+		{ID: "catalogStats.timeseries", Name: "timeseries"},
+		{ID: "catalogStats.views", Name: "views"},
+		{ID: "catalogStats.internalCollections", Name: "internalCollections"},
+		{ID: "catalogStats.internalViews", Name: "internalViews"},
+	},
+}
+
+var (
+	chartTcmallocGeneric = module.Chart{
+		ID:    "tcmallocGeneric",
+		Title: "Tcmalloc generic metrics",
+		Units: "MiB",
+		Fam:   "tcmalloc",
+		Ctx:   "mongodb.tcmallocGeneric",
+		Dims: module.Dims{
+			{ID: "tcmalloc.generic.current_allocated_bytes", Name: "current_allocated_bytes", Div: 1 << 20},
+			{ID: "tcmalloc.generic.heap_size", Name: "heap_size", Div: 1 << 20},
+		},
```

Comment:
```suggestion
	chartTcmallocGeneric = module.Chart{
		ID:    "tcmallocGeneric",
		Title: "Tcmalloc generic metrics",
		Units: "bytes",
		Fam:   "memory",
		Ctx:   "mongodb.tcmalloc_generic",
		Dims: module.Dims{
			{ID: "tcmalloc.generic.current_allocated_bytes", Name: "current_allocated_bytes"},
			{ID: "tcmalloc.generic.heap_size", Name: "heap_size"},
		},
```

- didn't find these metrics in [serverStatus](https://docs.mongodb.com/manual/reference/command/serverStatus/)
- I guess it makes sense to move it to "memory"
- it is in bytes, correct? If so there is no need to convert to something else
- "snake_case" Ctx

Reply:
I think that's one the non-documented features too. I had to trace it in the source code here https://github.com/mongodb/mongo/blob/master/src/mongo/util/tcmalloc_server_status_section.cpp